### PR TITLE
add placementdecision creating controller

### PR DESCRIPTION
**Pull request by elgnay**
Signed-off-by: Yang Le <yangle@redhat.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -26,4 +26,6 @@ GO_TEST_PACKAGES :=./pkg/...
 # $2 - Dockerfile path
 # $3 - context directory for image build
 # It will generate target "image-$(1)" for building the image and binding it as a prerequisite to target "images".
-$(call build-image,$(IMAGE),$(IMAGE_REGISTRY)/$(IMAGE),./Dockerfile,.)
\ No newline at end of file
+$(call build-image,$(IMAGE),$(IMAGE_REGISTRY)/$(IMAGE),./Dockerfile,.)
+
+include ./test/integration-test.mk
```
Comment (qiujian16) [inline Makefile:31]: eof

File: pkg/controllers/placementdecision/creating_controller.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+	}
+	labelSelector := labels.NewSelector().Add(*requirement)
+	placementDecisions, err := c.placementDecisionLister.PlacementDecisions(namespace).List(labelSelector)
+	if err != nil {
+		return err
+	}
+
+	// no work if PlacementDecision has been created
+	if len(placementDecisions) > 0 {
+		return c.updateOwnerReferences(ctx, placement, placementDecisions)
```
Comment (qiujian16) [inline pkg/controllers/placementdecision/creating_controller.go:110]: do we need to update owner here?

File: pkg/controllers/placementdecision/creating_controller.go [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+// updateOwnerReferences ensures all given placementDecisions have owner reference to the Placement
+func (c *placementDecisionCreatingController) updateOwnerReferences(ctx context.Context, placement *clusterapiv1alpha1.Placement, placementDecisions []*clusterapiv1alpha1.PlacementDecision) error {
+	errs := []error{}
+	for _, placementDecision := range placementDecisions {
+		if ok := metav1.IsControlledBy(placementDecision, placement); ok {
+			continue
+		}
+		placementDecision = placementDecision.DeepCopy()
+		owner := metav1.NewControllerRef(placement, clusterapiv1alpha1.GroupVersion.WithKind("Placement"))
```
Comment (qiujian16) [inline pkg/controllers/placementdecision/creating_controller.go:143]: a resource can only have one controllerref at one time.

File: Makefile (hunk 1 above)
Comment (elgnay) [reply Makefile]: Added.

File: pkg/controllers/placementdecision/creating_controller.go (hunk 2 above)
Comment (elgnay) [reply pkg/controllers/placementdecision/creating_controller.go]: Deleted the logic to update owner reference.

File: pkg/controllers/placementdecision/creating_controller.go (hunk 3 above)
Comment (elgnay) [reply pkg/controllers/placementdecision/creating_controller.go]: Code removed.

**Review by qiujian16 — COMMENTED** [review]
/lgtm

**Pull request merged**